runtime/lua/vim/lsp/rpc.lua

```diff
@@ -608,6 +608,73 @@ function M.connect(host, port)
   end
 end
 
+---@class Dispatchers
+---@field notification function
+---@field server_request function
+---@field on_exit function
+---@field on_error function
+
+-- TODO: better name
+---@class StartServer
+---@field request fun(method: string, params: table?, callbackfun: fun(err: lsp.ResponseError | nil, result: any), notify_reply_callback:function?)
+---@field notify fun(method: string, params: any)
+---@field is_closing function
+---@field terminate function
+
+--- Create a LSP RPC client factory that connects via named pipes (window)
+--- or unix domain sockets (unix) to the given pipe_path (file path on
+--- Unix and name on Windows)
+---
+---@param pipe_path  string
+---@return fun(dispatchers: Dispatchers): StartServer?
+function M.domain_socket_connect(pipe_path)
+  return function(dispatchers)
+    dispatchers = merge_dispatchers(dispatchers)
+    local pipe = uv.new_pipe(false)
+    if not pipe then
+      -- TODO: should an error be reported to the user?
+      return
+    end
+    local closing = false
+    local transport = {
+      write = vim.schedule_wrap(function(msg)
+        pipe:write(msg)
+      end),
+      is_closing = function()
+        return closing
+      end,
+      terminate = function()
+        if not closing then
```

Comment:
Should we handle the case where `closing` is `true`? (Might not be possible in theory but wild things happen).

Reply:
It looks like this is [only used to shutdown](https://github.com/TheLeoP/neovim/blob/0590f4d7c885889a721767d75ee60b7cb382f42d/runtime/lua/vim/lsp.lua#L1650-L1659) a LS. Should an error be log if `closing` is true? After a quick overview of [the spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#shutdown), I'm not sure about what would be the correct behaviour.

Reply:
If closing is already true then everything is fine. We don't need to turn a double-close into an error case.